azure/azjwt/azjwt.go

```diff
@@ -147,12 +148,24 @@ func (a *Auth) Authenticator(next http.Handler) http.Handler {
 			return
 		}
 
+		// Add the email to the context so that it can be used by the handler
+		ctx := context.WithValue(r.Context(), emailContextKey{}, allowedEmails[0])
 		// Token is authenticated, pass it through
-		next.ServeHTTP(w, r)
+		next.ServeHTTP(w, r.WithContext(ctx))
 	}
 	return http.HandlerFunc(hfn)
 }
 
+type emailContextKey struct{}
+
+func EmailFromContext(ctx context.Context) (string, error) {
```

Comment:
Normally, I'd suggest having this live in it's own package as to not introduce a tight coupling between azjwt + usersrv, but it really doesn't matter here

Reply:
GC - done.